SilKit/include/silkit/participant/IParticipant.hpp

```diff
@@ -99,6 +101,9 @@ class IParticipant
 
     //! \brief Return the ILogger at this SIL Kit participant.
     virtual auto GetLogger() -> Services::Logging::ILogger* = 0;
+
+    //! \brief Get a parameter set by an API call and/or the participant configuration.
+    virtual auto GetParameter(SilKit::Parameter parameter) -> const std::string& = 0;
```

Comment:
I would produce a new `std::string` on each invocation. This API shouldn't be called more than once (under normal circumstances) anyway. This will also simplify the implementation, because the caching in the Hourglass-API can be dropped.

```suggestion
    virtual auto GetParameter(SilKit::Parameter parameter) -> std::string = 0;
```